tests/unit/amo/components/TestReportUserAbuse.js

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+
+import ReportUserAbuse, {
+  ReportUserAbuseBase,
+} from 'amo/components/ReportUserAbuse';
+import { setError } from 'core/actions/errors';
+import {
+  hideUserAbuseReportUI,
+  loadUserAbuseReport,
+  sendUserAbuseReport,
+  showUserAbuseReportUI,
+} from 'amo/reducers/userAbuseReports';
+import DismissibleTextForm from 'ui/components/DismissibleTextForm';
+import ErrorList from 'ui/components/ErrorList';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import {
+  createFakeUserAbuseReport,
+  createStubErrorHandler,
+  createUserAccountResponse,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  const defaultRenderProps = {
+    errorHandler: createStubErrorHandler(),
+    i18n: fakeI18n(),
+    store: dispatchClientMetadata().store,
+    user: createUserAccountResponse({ id: 9 }),
+  };
+
+  function renderShallow({ ...props } = {}) {
+    return shallowUntilTarget(
+      <ReportUserAbuse {...defaultRenderProps} {...props} />,
+      ReportUserAbuseBase
+    );
+  }
+
+  it('renders a disabled button if no user exists', () => {
+    const root = renderShallow({ user: null });
+
+    expect(root.find('.ReportUserAbuse')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-show-more'))
+      .toHaveProp('disabled', true);
+  });
+
+  it('renders a button to report an user', () => {
+    const root = renderShallow();
+
+    expect(root.find('.ReportUserAbuse')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-show-more').prop('children'))
+      .toEqual('Report this user for abuse');
+  });
+
+  it('shows the preview content when first rendered', () => {
+    const root = renderShallow();
+
+    expect(root.find('.ReportUserAbuse--is-expanded')).toHaveLength(0);
+  });
+
+  it('shows more content when showReportUI is called', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const user = createUserAccountResponse();
+    const root = renderShallow({ store, user });
+
+    root.instance().showReportUI();
```

Comment:
Why not do `root.find('ReportUserAbuse-show-more').simulate('click', createFakeEvent())`? That would be more realistic and would cover the onClick logic.

Reply:
I think there was an issue with it not working, maybe because of shallow render? I'll try again and see...

Reply:
Oh actually, it was just that I needed to reach into the `DismissibleTextForm` component to do the hideUI action in the next test... but I can probably tidy this up. I'll give it a go...